folio-spring-base/src/main/java/org/folio/spring/client/EnrichUrlAndHeadersClient.java

```diff
@@ -23,26 +24,55 @@ public EnrichUrlAndHeadersClient(FolioExecutionContext folioExecutionContext, ok
 
   @Override
   public Response execute(Request request, Request.Options options) throws IOException {
-    String url;
-
-    var okapiUrl = folioExecutionContext.getOkapiUrl();
-    if (okapiUrl != null) {
-      if (!okapiUrl.endsWith("/")) {
-        okapiUrl += "/";
-      }
-      url = request.url().replace("http://", okapiUrl);
-    } else {
-      url = request.url();
+    String url = prepareUrl(request.url(), folioExecutionContext);
+
+    Map<String, Collection<String>> allHeaders = prepareHeaders(request, folioExecutionContext);
+
+    Request requestWithUrl = Request.create(
+      request.httpMethod(),
+      url,
+      allHeaders,
+      request.body(),
+      request.charset(),
+      request.requestTemplate()
+    );
+
+    log.debug(
+      "FolioExecutionContext: {};\nPrepared the Feign Client Request: {} with headers {};\nCurrent thread: {}",
+      folioExecutionContext,
+      requestWithUrl,
+      allHeaders,
+      Thread.currentThread().getName()
+    );
+
+    return delegate.execute(requestWithUrl, options);
+  }
+
+  protected static String prepareUrl(String requestUrl, FolioExecutionContext context) {
```

Comment:
Nit: I'd make this method package-private, to lock it down a little more (with the logic being that in shared libraries, you can always trivially make things more visible, while restricting visibility later on is nearly always potentially a breaking change)